src/Features/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -11,17 +11,22 @@
 
 namespace Microsoft.CodeAnalysis.Shared.TestHooks
 {
-    internal partial class AsynchronousOperationListener : IAsynchronousOperationListener, IAsynchronousOperationWaiter
+    internal sealed partial class AsynchronousOperationListener : IAsynchronousOperationListener, IAsynchronousOperationWaiter
     {
         private readonly object _gate = new object();
+        private readonly string _featureName;
         private readonly HashSet<TaskCompletionSource<bool>> _pendingTasks = new HashSet<TaskCompletionSource<bool>>();
+
         private List<DiagnosticAsyncToken> _diagnosticTokenList = new List<DiagnosticAsyncToken>();
         private int _counter;
         private bool _trackActiveTokens;
 
-        public AsynchronousOperationListener()
+        public AsynchronousOperationListener(string featureName = "noname", bool tracking = false)
```

Comment:
Should we call this parameter `enableDiagnosticTokens` or something? Since this is really a tri-state conceptually (no tracking period, tracking with cheap tokens, tracking with expensive tokens), and I wasn't sure which this meant.